### PR TITLE
Update instructions for working with ghc-lib

### PR DESCRIPTION
**Pull request by nickchapman-da**
Update instructions for working with ghc-lib

- Describe how to use a windows ad-hoc machine to run the `bazel` pin step.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thanks!

File: ghc-lib/new-working-on-ghc-lib.md [hunk 1]
```diff
@@ -65,3 +65,91 @@ Once you have the GHC patch you want to incorporate into the DAML repo, here's t
...
+```
+
+2. Create a new windows machine:
+```
+./ad-hoc.sh create windows
+```
+
+3. When the script complete, note the IP and Password.
+
+4. Wait at least 10 minutes before trying to remotely connect to the windows machine, or else it may not inialize correctly. Really, I am not kidding. This advice is from Gary!
```
Comment (garyverhaegen-da) [inline ghc-lib/new-working-on-ghc-lib.md:86]: Unfortunately true. I'll try to find some time to investigate further over the next few days, but it seems really bizarre so far.

File: ghc-lib/new-working-on-ghc-lib.md [hunk 2]
```diff
@@ -65,3 +65,91 @@ Once you have the GHC patch you want to incorporate into the DAML repo, here's t
...
+12. Edit a powershell script: Remove all calls to the bazel function at the end of the file:
+```
+vi build.ps1
+```
+
+13. Run the powershell script. It uses `Scoop` to install stuff. Takes about 5 minutes. Say "yes" to a couple of popup dialogues:
+```
+$ powershell
+PS C:\Users\u\daml> .\build.ps1
+PS C:\Users\u\daml> exit
```
Comment (garyverhaegen-da) [inline ghc-lib/new-working-on-ghc-lib.md:134]:
```suggestion
```

I don't think exiting PowerShell works here. At least when running through it today I believe you stayed in PowerShell and ran all of the following commands from there.

File: ghc-lib/new-working-on-ghc-lib.md [hunk 3]
```diff
@@ -65,3 +65,91 @@ Once you have the GHC patch you want to incorporate into the DAML repo, here's t
...
+```
+bazel run @stackage-unpinned//:pin
+```
+
+15. Commit the change, and push upstream:
+```
+git add .\stackage_snapshot_windows.json
+git config user.email "you@example.com""
+git config user.name "Your Name"
+git commit -m 'update snapshot after pin on windows'
```
Comment (garyverhaegen-da) [inline ghc-lib/new-working-on-ghc-lib.md:147]: Possibly shorter:
```
git -c "user.email=..." -c "user.name=..." commit -m "...."
```

**Review by garyverhaegen-da — APPROVED** [review]

File: ghc-lib/new-working-on-ghc-lib.md (hunk 2 above)
Comment (nickchapman-da) [reply ghc-lib/new-working-on-ghc-lib.md]: thank you. good spot. we did stay in `powershell`

**Pull request merged**